logstash-core/lib/logstash/webserver.rb

```diff
@@ -0,0 +1,66 @@
+# encoding: utf-8
+require "puma"
+require 'puma/single'
+require 'puma/binder'
+require 'puma/configuration'
+
+class LogStash::WebServer::Base
+
+  attr_reader :logger, :status, :config, :options
+
+  def initialize(logger, options={})
+    @logger  = logger
+    @options = options
+    @status  = nil
+
+    set_environment
+  end
+
+  def run
+    raise "Not implemented"
+  end
+
+  def log(str)
+    logger.debug(str) if logger.debug?
+  end
+
+  def stop(options={})
+    raise "Not implemented"
+  end
+
+
+  def setup_signals
+    Signal.trap("INT") do
+      @status = :exit
+      stop(:graceful => true)
+      exit
+    end
+
+    begin
+      Signal.trap "SIGTERM" do
+        stop
+      end
+    rescue Exception
+      log "*** SIGTERM not implemented, signal based gracefully stopping unavailable!"
+    end
+  end
```

Comment:
This is already handler inside the `agent.rb` I don't think we need to have a custom logic in our webserver